### PR TITLE
Update integrations.md

### LINKED ISSUE #34: Discoverability of inline PR comment feature in docs
Right now all the information around inline pr comments is just in a comment in an example yaml file. It would be easier to find if it had its own header.

(After making these changes, verify that searching the docs for inline pr comments will return a relevant result, which it doesn't do currently)

### PR DESCRIPTION
**Pull request by sabihb**
Fixes #34

### CODE REVIEW

File: docs/integrations.md [hunk 1]
```diff
@@ -88,6 +85,11 @@ jobs:
       #  if: always()
 ```
 
+#### Inline PR Comments (beta)
+Set `GITHUB_TOKEN` (commented out in the code above) to leave inline comments on your pull requests.
```
Comment (dlukeomalley) [inline docs/integrations.md:89]: @sabihb Could we add a `!!! info` notice that this feature is currently for GitHub users only?

When you say "commented out in the code above", what code above has it commented out? Do all of the providers have that code?

File: docs/integrations.md [hunk 2]
```diff
@@ -88,6 +85,11 @@ jobs:
...
+Comments are left when Semgrep CI finds a result that blocks CI.
```
Comment (dlukeomalley) [inline docs/integrations.md:90]: This surfaces a good product question! Should inline comments only be triggered by "block CI" results? I'd love to chat this week, I've gotten feature requests for notify-only WITH inline pr comments :)

**Review by dlukeomalley — CHANGES_REQUESTED** [review]

File: docs/integrations.md (hunk 2 above)
Comment (sabihb) [reply docs/integrations.md:90]: That is a very good question - I wonder which is more intuitive for the user. I think the reasoning for implementing it firing on "Block CI" is that an inline PR comment does "block" a PR.

Also, what exactly is "block CI" for in general? Findings that should cause CI to fail? (In which case, are rules/rulesets specified to not block CI effectively disabled?)

File: docs/integrations.md (hunk 1 above)
Comment (sabihb) [reply docs/integrations.md:89]: 👍  Will add the notice! I'll clarify the sentence about the commented out code. I was attempting to refer to the yaml file example above this paragraph (where the two lines to add inline pr comments are commented out).

File: docs/integrations.md (hunk 2 above)
Comment (dlukeomalley) [reply docs/integrations.md:90]: Does an inline PR comment have to block a PR or can it be informational (i.e. done outside of a PR-review)? Rules / rulesets specified not to block CI are effectively informational.

Comment (sabihb) [reply docs/integrations.md:94]: Just looking around, it seems you can "Add single comment", which doesn't require a review. The comments right now do start a review (which is not something I remember setting, so it must just be the endpoint we're using), but I'm sure there exists an API endpoint that allows you to just leave an inline comment.

File: docs/integrations.md (hunk 1 above)
Comment (sabihb) [reply docs/integrations.md]: Updated! Let me know what you think.

**Review by dlukeomalley — APPROVED** [review]

**Pull request merged**